### PR TITLE
Cfy 3146 aws system tests handler cleanup implementation

### PR DESCRIPTION
**Pull request by gilzellner**

### CODE REVIEW

File: system_tests/ec2_handler.py [hunk 1]
```diff
@@ -51,6 +54,20 @@ def get_resources_to_teardown(self):
         return self.env.handler.ec2_infra_state_delta(
             before=self.before_run, after=current_state)
 
+    @classmethod
+    def clean_all(cls, env):
+        resources_to_be_removed = env.handler.ec2_infra_state()
+        cls.logger.info("Current resources in account: " + str(resources_to_be_removed))
```
Comment (dankilman) [inline system_tests/ec2_handler.py]: use format instead of + str(WHATEVER)

File: system_tests/ec2_handler.py [hunk 2]
```diff
@@ -51,6 +54,20 @@ def get_resources_to_teardown(self):
...
+        cls.logger.info("Hardcoded resources to not remove: " + str(KEYS_TO_NOT_REMOVE_ON_CLEANUP))
+        resources_to_be_removed['key_pairs'].pop(KEYS_TO_NOT_REMOVE_ON_CLEANUP)
+        cls.logger.info("resources_to_be_removed: " + str(resources_to_be_removed))
+        failed = env.handler.remove_ec2_resources(resources_to_be_removed)
+        assert len(failed['instances']) == 0
```
Comment (dankilman) [inline system_tests/ec2_handler.py]: do `if`'s and if assertion fails, throw meaningful exception with proper message

File: system_tests/ec2_handler.py [hunk 3]
```diff
@@ -51,6 +54,20 @@ def get_resources_to_teardown(self):
...
+        assert len(failed['key_pairs']) == 0
+        assert len(failed['elasticips']) == 0
+        assert len(failed['security_groups']) == 1
+        # This is the default security group which cannot be removed by a user.
```
Comment (dankilman) [inline system_tests/ec2_handler.py]: the comment should be placed before the security group condition and not after it

File: system_tests/ec2_handler.py (hunk 1 above)
Comment (gilzellner) [inline system_tests/ec2_handler.py]: done

File: system_tests/ec2_handler.py (hunk 2 above)
Comment (gilzellner) [inline system_tests/ec2_handler.py]: done

File: system_tests/ec2_handler.py (hunk 3 above)
Comment (gilzellner) [inline system_tests/ec2_handler.py]: done

File: system_tests/ec2_handler.py [hunk 4]
```diff
@@ -98,6 +120,22 @@ def agents_security_group(self):
     def management_security_group(self):
         return self.config['mananger_security_group_name']
 
+    @property
+    def manager_keypair_name(self):
```
Comment (dankilman) [inline system_tests/ec2_handler.py]: remove these

**Pull request merged**